runtime/service/npud/backend/trix/TrixBackend.cc

```diff
@@ -307,8 +307,105 @@ NpuStatus TrixBackend::setRequestData(NpuContext *ctx, RequestID requestId, Inpu
                                       TensorDataInfos *inputInfos, OutputBuffers *outputBufs,
                                       TensorDataInfos *outputInfos)
 {
-  // TODO Implement details
-  return NPU_STATUS_ERROR_NOT_SUPPORTED;
+  auto citer = std::find_if(_dev->ctxs.begin(), _dev->ctxs.end(),
+                            [&](std::unique_ptr<NpuContext> &c) { return c.get() == ctx; });
+  if (citer == _dev->ctxs.end())
+  {
+    return NPU_STATUS_ERROR_INVALID_ARGUMENT;
+  }
+
+  auto riter = std::find(ctx->requests.begin(), ctx->requests.end(), requestId);
+  if (riter == ctx->requests.end())
+  {
+    return NPU_STATUS_ERROR_INVALID_ARGUMENT;
+  }
+
+  auto &req = _dev->requests.at(requestId);
+  auto miter = std::find(ctx->models.begin(), ctx->models.end(), req->modelId);
+  if (miter == ctx->models.end())
+  {
+    return NPU_STATUS_ERROR_INVALID_MODEL;
+  }
```

Comment:
Can't we use try-catch from `std::map::at(...)`'s `std::out_of_range` exception?

Reply:
Um, before calling `at` function, it already checks whether the `requestId` is included in ctx requests array. Still, should I use `try-catch` phrase?

Reply:
Oh, do you mention it because of my TODO comment?

Reply:
>Um, before calling `at` function, it already checks whether the `requestId` is included in ctx requests array. Still, should I use `try-catch` phrase?

My comment means that we can remove `std::find(...)` checking if we use exception.

Reply:
@hseok-oh 
Um, I can't find a way to declare the reference variable of the unique_ptr item outside the `try-catch` block. I don't want to use `get()` function of unique_ptr... Could you suggest an elegant way to do this?

```
  TrixRequestInfo *req;
  try
  {
    req = _dev->requests.at(requestId).get();
  }
  catch(const std::exception &e)
  {
    return NPU_STATUS_ERROR_INVALID_ARGUMENT;
  }
```

Reply:
I also don't have any idea for elegant implementation. :)